File: server/ncurses/application.php
<?php

require_once('ncurses/screen.php');


class Application {
    private static $screen;
    private static $widgets = array();

    function __construct() {
        $this->screen = new Screen();
    }

    static function add($widget) {
        array_push($this->widgets, $widget);
    }

    function draw() {
        $this->screen->clear();
        foreach ($this->widgets as $widget) {
            $widget->draw($this->screen);
        }
    }

    function getScreen() {
        return $this->screen;
    }

}

?>

File: server/ncurses/application.py
from ncurses.label import Label
from ncurses.screen import Screen
from ncurses.dialog import Dialog
from ncurses.button import Button


class Application:
    widgets = []

    def __init__(self):
        self.screen = Screen()
        pass

        movingLabel = Label("HELLO WORLD!", 2, 1)
        self.add(movingLabel)
        dialog1 = Dialog("DIALOG", 10, 10, 21, 11)
        self.add(dialog1)
        dialog1.add(Button("OK", 2, 2))
        dialog1.add(Button("CANCEL", 8, 2))
        movingLabel.setText('MEDLIK')
        self.draw()


    def add(self, widget):
        self.widgets.append(widget)
    

    def draw(self):
        self.screen.clear()
        for widget in self.widgets:
            widget.draw(self.screen)
        
